### ISSUE TITLE
NRE thrown during add-migration if primary key is null.

### ISSUE THREAD

**Issue by mrswain** [aspnet/EntityFrameworkCore#10571]
Other code in migration generators seem to handle the absence of a primary key, but this line

https://github.com/aspnet/EntityFrameworkCore/blob/69f870ab1f32b3185c3599887fbe020c0f70c445/src/EFCore.Design/Migrations/Design/MigrationsCodeGenerator.cs#L129

does not perform a null check and will pass a null back to the annotations collector thus firing a null pointer exception.

Given that other areas of the code generator now properly check for and handle a null for PrimaryKey on CreateTableOperation, I figure this was just overlooked due to code complexity.

### Further technical details
EF Core version: 2.1.0-preview1-27746
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Win10
IDE:  Visual Studio 2017 15.5

**Comment by ajcvickers** [comment]
@mrswain Can you provide some details on how you are hitting this? Entity types in EF Core models must have primary keys defined, and model validation should catch this before reaching this code.

**Comment by mrswain** [comment]
Yes, they must have PKs, but the structure I am working with is a dataset pushed nightly into SQL from a DB2 source, and it has no defined primary key. 

To work around this in the entity I defined a best guess PK but since I cannot ensure that the constraint would be truly valid, since it does not exist on the source, I am attempting to convert it to a clustered index. I have a custom annotation on the entity to denote this, but pushing the annotation down to the migration always looked ugly.  I saw that the SQL server provider will check for the PK and only use it if it exists, so I was attempting to intercept the code generator and create a clustered index instead of a PK. That is when I encountered the null reference exception and traced it to the Annotation collector. 

Since the SQL provider does a null check against the PK I thought that this may just have been an oversight and wanted to bring it to your attention. I understand that this is probably a wont-fix/as-designed situation but adding a if != null check before the marked line is a simple fix and should not cause any breakage.
